### PR TITLE
Impl 1760

### PR DESCRIPTION
**Pull request by DeanPalmer**
Created the scrollIntercepter to handle requests to scroll the viewport. These were the only directives found that were scrolling, please let me know if I have missed any.

### CODE REVIEW

File: src/core/javascripts/directives/main/bb_scroll_to.directive.js [hunk 1]
```diff
@@ -35,13 +35,7 @@ angular.module('BB.Directives').directive('bbScrollTo', ($rootScope, AppConfig,
...
                         if (((evnt === "page:loaded") && (current_step > 1)) || always_scroll || (evnt === "widget:restart") ||
                             (!isElementInView(scroll_to_element) && (scroll_to_element.offset().top !== 0))) {
-                            if ('parentIFrame' in $window) {
-                                return parentIFrame.scrollToOffset(0, scroll_to_element.offset().top - GeneralOptions.scroll_offset);
-                            } else {
-                                return $bbug("html, body").animate(
-                                    {scrollTop: scroll_to_element.offset().top - GeneralOptions.scroll_offset}
-                                    , bb_transition_time);
-                            }
+                            scrollIntercepter.scrollToElement(scroll_to_element, bb_transition_time, evnts);
```
Comment (lukeferguson) [inline src/core/javascripts/directives/main/bb_scroll_to.directive.js]: Can you just pass `evnt` instead? Do you need all of them?

File: src/core/javascripts/services/scrollIntercepter.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+ * Accepts requests to scroll the viewport, depending on the type the request is either fullfilled or rejected.
+ *
+ */
+
+angular.module('BB.Services').factory('scrollIntercepter', ($bbug, $window, GeneralOptions, $timeout) => {
+
+    var currentlyScrolling = false;
+
+    var scrollToElement = function(element, transitionTime, type) {
+        if (typeof(type) === 'object') {
```
Comment (lukeferguson) [inline src/core/javascripts/services/scrollIntercepter.js]: Assuming this is to handle the array data type? based on earlier comment, might be able to get rid of this!

**Comment by DeanPalmer** [comment]
scrollintercepter

This Gif shows the intercepter allowing an alert:raised event to take precedence over other scroll events and shows the time:selected request being ignored as an form:invalid request is already happening.

File: src/core/javascripts/directives/main/bb_scroll_to.directive.js (hunk 1 above)
Comment (DeanPalmer) [inline src/core/javascripts/directives/main/bb_scroll_to.directive.js]: Yeah you are correct, evnt is what we are looking for

File: src/core/javascripts/services/scrollIntercepter.js (hunk 2 above)
Comment (DeanPalmer) [inline src/core/javascripts/services/scrollIntercepter.js]: Will remove.

**Pull request merged**